torch/jit/_recursive.py

```diff
@@ -377,6 +377,15 @@ def init_fn(script_module):
         torch._C._run_emit_module_hook(cpp_module)
         concrete_type_store.methods_compiled.add(concrete_type)
 
+    # Special handling so methods like __len__ work in script methods on classes derived from containers
+    if isinstance(nn_module, (torch.nn.ModuleList, torch.nn.Sequential, torch.nn.ModuleDict)) and \
```

Comment:
maybe worth checking that the module doesn't override the ipmlementation in the `ModuleList`/`Sequential`/`ModuleDict`

Reply:
what do you mean? i think i actually handled this case correctly, although implicitly, by doing this block of code right after the '# Compile methods if necessary' block.  IIUC, if they override __len__ it gets compiled in that block, then my block sees __len__ exist and bails.  I did confirm this in one particular test case which overrides __len__.

Reply:
I think of someone overrides __len__ but doesn’t add @export to it, it won’t be present on the cpp module when we do the check on the cpp module and we’ll still do the custom compilation